Sources/Components/Broadcast/BroadcastContainerView/BroadcastContainerView.swift

```diff
@@ -0,0 +1,258 @@
+        }
+
+        struct BroadcastView {
+            static let spacing: CGFloat = .mediumLargeSpacing
+        }
+    }
+
+    /// The datasource for the BroadcastContainerView
+    public weak var dataSource: BroadcastContainerViewDataSource? {
+        didSet {
+            reload()
+        }
+    }
+
+    /// The delegate of the BroadcastContainerView
+    public weak var delegate: BroadcastContainerViewDelegate?
+
+    private lazy var contentView: UIStackView = {
+        let view = UIStackView(frame: .zero)
+
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.axis = .vertical
+        view.spacing = LayoutConstants.BroadcastView.spacing
+        view.distribution = .fillProportionally
+
+        return view
+    }()
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    public override func didMoveToSuperview() {
+        contentView.frame = CGRect(x: 0, y: 0, width: frame.width, height: 0)
+    }
+}
+
+// MARK: - Public
+
+public extension BroadcastContainerView {
+    /// Reloads the container by removing existing broadcasts and laying out all the broadcasts that the datasource provides
+    func reload() {
+        guard let dataSource = dataSource else {
+            return
+        }
+
+        reset()
+
+        let rangeOfBroadcastsToDisplay = 0 ..< dataSource.numberOfBroadcastsToDisplay(in: self)
+        let broadcastToDisplay = rangeOfBroadcastsToDisplay.map { dataSource.broadcastContainerView(self, broadcastForIndex: $0) }
+
+        if broadcastToDisplay.isEmpty {
+            return
+        }
+
+        layoutBroadcastViews(from: broadcastToDisplay)
+
+        if let delegate = delegate {
+            delegate.broadcastContainer(self, willDisplayBroadcastsWithContainerSize: intrinsicContentSize, commitToDisplaying: {
+                UIView.animate(withDuration: 0.3, animations: { [weak self] in
+                    self?.contentView.arrangedSubviews.forEach { $0.isHidden = false }
+                })
+            })
+        } else {
+            UIView.animate(withDuration: 0.3, animations: { [weak self] in
+                self?.contentView.arrangedSubviews.forEach { $0.isHidden = false }
+            })
+        }
+    }
+
+    public override var intrinsicContentSize: CGSize {
+        guard contentView.arrangedSubviews.count != 0 else {
+            return CGSize(width: frame.size.width, height: 0)
+        }
+
+        let broadcastViewsHorizontalSpacings = LayoutConstants.ContentView.insets.leading + abs(LayoutConstants.ContentView.insets.trailing)
+        let constrainedWidth = frame.size.width - broadcastViewsHorizontalSpacings
+
+        let broadcastViewsTotalHeight = contentView.arrangedSubviews.reduce(CGFloat(0), { acc, view in
```

Comment:
What's acc? Could we make this more verbose?

Reply:
accumulator, hehe. I'll make it more verbose 👍